### PR TITLE
did the hw by changing projectile frame and adding an interface

### PR DESCRIPTION
**Pull request by hklerer**
Screenshot 2024-02-21 at 6 57 04 PM

### CODE REVIEW

File: src/main/java/klerer/projectile/ProjectileFrame.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    public ProjectileFrame() {
+        setSize(400, 600);
+        setTitle("Projectile Calculator");
+        setDefaultCloseOperation(EXIT_ON_CLOSE);
+
+        setLayout(new GridLayout(8, 2));
+        JLabel velocityLabel = new JLabel("Velocity");
+        JLabel angleLabel = new JLabel("Angle");
+
+        final int ANGLE_MIN = 0;
```
[the remark this answers is not in the document]
Comment (andrewoid) [reply src/main/java/klerer/projectile/ProjectileFrame.java]: Make these static constants and move them out of this constructor to get rid of this error.

File: src/main/java/klerer/projectile/ProjectileFrame.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                        Double.parseDouble(velocityField.getText()),
+                        Double.parseDouble(String.valueOf(angleSlider.getValue()))
+                );
+                projectile.setSeconds(
+                        Double.parseDouble(secondsField.getText())
+                );
+                calculatedXLabel.setText(Double.toString(projectile.getX()));
+                calculatedYLabel.setText(Double.toString(projectile.getY()));
+                peakY.setText(Double.toString(projectile.getPeakY()));
+                interceptX.setText(Double.toString(projectile.getInterceptX()));
```
Comment (andrewoid) [inline src/main/java/klerer/projectile/ProjectileFrame.java]: Remove duplicated code.

File: src/main/java/klerer/projectile/ProjectileFrame.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        add(empty);
+        add(calculateButton);
+
+
+        velocityField.getDocument().addDocumentListener(new SimpleDocumentListener() {
+            @Override
+            public void update(DocumentEvent e) {
+                Projectile projectile = new Projectile(
+                        Double.parseDouble(velocityField.getText()),
+                        Double.parseDouble(String.valueOf(angleSlider.getValue()))
```
Comment (andrewoid) [inline src/main/java/klerer/projectile/ProjectileFrame.java]: These are backwards, it should be angle then velocity.

Comment (andrewoid) [reply src/main/java/klerer/projectile/ProjectileFrame.java]: Also `angleSlider.getValue()` can be used without parseDouble() or valueOf().
You're taking an integer, turning it into a String just to turn it into a double.

File: src/main/java/klerer/projectile/ProjectileFrame.java (hunk 2 above)
Comment (hklerer) [reply src/main/java/klerer/projectile/ProjectileFrame.java]: so what goes inside the update?

Comment (andrewoid) [reply src/main/java/klerer/projectile/ProjectileFrame.java]: I mean you have the same code copy and pasted to multiple places.

Only write it once.

Comment (hklerer) [reply src/main/java/klerer/projectile/ProjectileFrame.java]: so I should make a method out of this code?

Comment (andrewoid) [reply src/main/java/klerer/projectile/ProjectileFrame.java]: That would solve the issue

Comment (hklerer) [reply src/main/java/klerer/projectile/ProjectileFrame.java]: does the method go outside projectileFrame() and then I pass everything as parameters or do I make it inside the projectileFrame()?

Comment (andrewoid) [reply src/main/java/klerer/projectile/ProjectileFrame.java]: You do not need permission to write code.

Comment (hklerer) [reply src/main/java/klerer/projectile/ProjectileFrame.java]: its getting errors when i do it

Comment (andrewoid) [reply src/main/java/klerer/projectile/ProjectileFrame.java]: Then you aren't doing it right

**Review by andrewoid — APPROVED** [review]

**Pull request merged**